pkg/services/datasources/models.go

```diff
@@ -77,6 +79,18 @@ type TeamHTTPHeader struct {
 	Value  string `json:"value"`
 }
 
+// TeamHTTPHeaderValueRegexMatch returns a regex that can be used to check
+// words separated by special characters
+// "namespace!=auth", "env="prod", "env!~dev"
+func (th *TeamHTTPHeader) TeamHTTPHeaderValueRegexMatch() (*regexp.Regexp, error) {
+	exp := `^[\w\-\.\*\!\~\=]+$`
```

Comment:
I'm not sure this is correct regex for this problem. Why `.` and `*` here?. We have expressions like `label (= | != | =~ | !~) "value" [, ...]`. Pay attention to that value should be quoted. I would say something like this works better:
```
^(?:\s*\w+(?:=|!=|=~|!~)\"\w+\",*)+$
```

Reply:
we could copy the label matcher they have in gcom. found this

https://github.com/grafana/cloud-features/blob/c7e9abd67983768bd9e0005a689236d58d95f982/src/features/access-policies/utils/labels.ts#L8-L27